lib/einhorn/command.rb

```diff
@@ -123,10 +123,14 @@ def self.signal_all(signal, children=nil, record=true)
         end
       end
 
-      if Einhorn::State.signal_timeout
+      if Einhorn::State.signal_timeout && record
         Einhorn::Event::Timer.open(Einhorn::State.signal_timeout) do
           children.each do |child|
             next unless spec = Einhorn::State.children[child]
+            unless spec[:signaled].include?(signal)
```

Comment:
This is still potentially racy if we have two restarts within the signal_timeout window, right?

- Restart 1 signals PID and schedules callback
- PID dies and is replaced by equivalent PID
- Restart 2 signals same PID (but new process) and schedules 2nd callback
- First callback fires and new process is killed after less than the signal timeout.